spec.md

```diff
@@ -144,18 +144,51 @@ circuit MyTop...
 
 ## Circuits
 
-All FIRRTL circuits consist of a list of modules, each representing a hardware
-block that can be instantiated. The circuit must specify the name of the
-top-level module.
+A FIRRTL circuit is a named collection of FIRRTL modules.  Each module is a
+hardware "unit" that has ports, registers, wires, and may instantiate other
+modules (see: [@sec:modules]).  (This is the same concept as a Verilog
+`module`{.verilog}.)  A public module is usable outside of the current circuit.
+
+Consider the following circuit named `Foo`.  This contains two modules, `Bar`
+and `Baz`.  Module `Baz` is marked public.
 
 ``` firrtl
-circuit MyTop :
-  module MyTop :
-    ; ...
-  module MyModule :
-    ; ...
+circuit Foo :
+  module Bar :
+    input a: UInt<1>
+    output b: UInt<1>
+
+    connect b, a
+
+  public module Bar :
+    input a: UInt<1>
+    output b: UInt<1>
+
+    inst bar of Bar
+    connect bar.a, a
+    connect b, bar.b
+```
+
+A circuit may contain public modules that instantiate no common instances.
```

Comment:
Saying same thing you said a few different ways but just to be clear:

Dropping "main module".  Circuits are kinda like packages of public entirely disjoint circuits.
Is that right?

1) Can they instantiate the same extmodule's? Or is the rule about not sharing any instances only for definitions within the circuit?
2) Can they share anything at all, like the recent top-level definitions (class/extclass, properties, type alias, so on)?
3) Other than the filelist detail, I think what's proposed here is very much like what you'd get with different circuits (each with own "top") but instead in same file-- with difference being moving multiple circuits into same circuit requires them to be disjoint (where-as today multiple circuits can of course extmodule and use the other).  IS that about right?

Reply:
i think you read the eats shoots and leaves like i did ... the statement is that they are *allowed* to instantiate the same (non-public) modules and extmodules. They do not HAVE to share any commmon modules and extmodules.

In other words, (https://mentor.ieee.org/myproject/Public/mytools/draft/styleman.pdf)

- public modules `MAY` share common modules and extmodules.
- public modules `SHALL NOT` be instantiated by any other modules.

Reply:
Oh my goodness, totally! Hahaha, thanks!

D'oh!

Thanks for the `MAY` / `SHALL NOT` language :+1: . Okay, great.  That makes a lot of sense.

Reply:
There's no requirement on disjointness. This statement was more to make it clear that the public modules _may_ have entirely disjoint instance graphs. However, they may also share anything. Put differently, circuits are collections of `public` modules. What those `public` modules instantiate or reference, has no real bearing on anybody using them as the rest of this isn't visible via the ABI.